python/ray/plasma/plasma.py

```diff
@@ -46,6 +47,16 @@ def start_plasma_store(plasma_store_memory=DEFAULT_PLASMA_STORE_MEMORY,
     """
     if use_valgrind and use_profiler:
         raise Exception("Cannot use valgrind and profiler at the same time.")
+
+    if huge_pages_enabled and not (sys.platform == "linux" or
+                                   sys.platform == "linux2"):
+        raise Exception("The huge_pages_enabled argument is only supported on "
+                        "Linux.")
+
+    if huge_pages_enabled and plasma_directory is None:
```

Comment:
we have a check for this in plasma store.cc
```
  if (hugepages_enabled && plasma_directory.empty()) {
    ARROW_LOG(FATAL) << "if you want to use hugepages, please specify path to huge pages "
                        "filesystem with -d";
  }
```
I suppose it doesn't hurt to catch it a bit earlier.

Reply:
That check isn't sufficient because it will crash the object store. Errors need to result in Python exceptions.